### ISSUE TITLE
Conversions are not generated with custom_path_generator_class

### ISSUE THREAD

**Issue by loranger** [spatie/laravel-medialibrary#233]
Hello,

I use a custom class for my media storage, here it is :

```php
<?php

namespace App\Services;

use Spatie\MediaLibrary\PathGenerator\PathGenerator;
use Spatie\MediaLibrary\Media;

class MediaPathGenerator implements PathGenerator
{

    /**
    * Build the path for a given media id.
    *
    * @param integer $id
    *
    * @return string
    */
    private function getSubPathFromID($id)
    {
        return implode(str_split(substr(sprintf('%05d', $id), -5)), '/') . '/' . $id;
    }

    /**
     * Get the path for the given media, relative to the root storage path.
     *
     * @param \Spatie\MediaLibrary\Media $media
     *
     * @return string
     */
    public function getPath(Media $media) :string
    {
        return $this->getSubPathFromID($media->id).'/';
    }

    /**
     * Get the path for conversions of the given media, relative to the root storage path.
     *
     * @param \Spatie\MediaLibrary\Media $media
     *
     * @return string
     */
    public function getPathForConversions(Media $media) :string
    {
        return $this->getSubPathFromID($media->id).'/c/';
    }
}
```
... and it work very well, except for the conversions.

The original file is stored at the right place, my model can retrieve the right urls :

```bash
Psy Shell v0.7.2 (PHP 7.0.7 — cli) by Justin Hileman
>>> \App\Link::find(2)->getMedia()[0]->getUrl()
=> "/media/0/0/0/0/1/1/WrpxLpXq.jpg"
>>> \App\Link::find(2)->getMedia()[0]->getUrl('thumb')
=> "/media/0/0/0/0/1/1/c/thumb.jpg"
>>>
```

... But the thumb.jpg file is never created (and the `c` folder either).

Even if the regeneration seems correct, the conversions are not created.

```bash
art medialibrary:regenerate                                             ⏎ master ✱
Media 1 regenerated
Media 2 regenerated
Media 3 regenerated
Media 4 regenerated
...
Media 110 regenerated
Media 111 regenerated
Media 112 regenerated
All done!
```

Did I miss something ?

**Comment by freekmurze** [comment]
Are you using queues to generate the conversions? If so are the queues running?

Are the conversions creating if you remove your custom path generator class?

**Comment by loranger** [comment]
Hello,

That was it : I did setup a queue worker after implementing medialibrary and did not notice medialibrary automatically uses it whereas my queue is not running as I'm doing tests.

My apologies, thank you for your answer.

By the way, while implementing my custom path generator, I noticed the documentation is not up-to-date regarding the interface's methods signatures. The implementation now requires any “return type”. I've just [opened a PR](https://github.com/spatie/docs.spatie.be/pull/25) regarding this documentation typo.
